src/libre/ast_analysis.c

```diff
@@ -492,23 +514,27 @@ assign_lasts(struct ast_expr *n)
 	case AST_EXPR_ALT: {
 		size_t i;
 
-		set_flags(n, AST_EXPR_FLAG_LAST);
+		set_flags(n, AST_FLAG_LAST);
 		for (i = 0; i < n->u.alt.count; i++) {
 			assign_lasts(n->u.alt.n[i]);
 		}
 		break;
 	}
 
 	case AST_EXPR_REPEATED:
-		set_flags(n, AST_EXPR_FLAG_LAST);
+		set_flags(n, AST_FLAG_LAST);
 		assign_lasts(n->u.repeated.e);
 		break;
 
 	case AST_EXPR_GROUP:
-		set_flags(n, AST_EXPR_FLAG_LAST);
+		set_flags(n, AST_FLAG_LAST);
 		assign_lasts(n->u.group.e);
 		break;
 
+	case AST_EXPR_SUBTRACT:
+		/* XXX: not sure */
+		break;
+
```

Comment:
@silentbicycle also this

Reply:
It seems like this and the corresponding case on line 471 should set the FIRST/LAST flags and recurse to their contents, like with REPEATED/GROUP, but I'm struggling to come up with a test case that proves that it matters. It may be a non-issue because they can only ultimately contain non-nullable items? And input regexes like `(^a|[^[:digit:]]$)c$` seem to handle pruning unsatisfiable parts correctly.